src/SchemaParser.f90

```diff
@@ -356,7 +356,7 @@ SUBROUTINE init_SchElm(this,name,pListPath,type,required)
   INTEGER(SIK),INTENT(IN) :: type
   LOGICAL(SBK),INTENT(IN) :: required
 
-  REQUIRE(type == SCHEMA_ELEMENT_REQUIRED .OR. type == SCHEMA_ELEMENT_NOT_REQUIRED)
+  REQUIRE(required == SCHEMA_ELEMENT_REQUIRED .OR. required == SCHEMA_ELEMENT_NOT_REQUIRED)
```

Comment:
Hm, this is odd... you changed from integer to logical without changing the types of the parameters you're comparing against?

Reply:
This should be fixed now